src/main/java/org/wso2/carbon/inbound/amazonsqs/AmazonSQSPollingConsumer.java

```diff
@@ -413,4 +413,18 @@ public void destroy() {
             logger.error("Error while shutdown the AmazonSQS " + name + " " + e.getMessage(), e);
         }
     }
-}
\ No newline at end of file
+
+    /**
+     * Resume the connection to the Amazon SQS.
+     */
+    public void resume() {
+        isConnected = false;
+    }
+
+    /**
+     * Pause the connection to the Amazon SQS.
+     */
+    public void pause() {
+        destroy();
```

Comment:
If the destroy method is called half way through the message processing, there will be some errors. We need to complete all the inflight messages before making the inbound endpoint pause. Can you confirm that behavior?